biorbd_optim/penalty.py

```diff
@@ -169,6 +169,38 @@ def minimize_predicted_com_height(penalty_type, ocp, nlp, t, x, u, **extra_param
 
                 penalty_type._add_to_penalty(ocp, nlp, CoM_height, **extra_param)
 
+        @staticmethod
+        def minimize_contact_forces(penalty_type, ocp, nlp, t, x, u, contacts_idx=(), data_to_track=(), **extra_param):
+            from casadi import Function
+            from .dynamics import Dynamics
+
+            CS_func = Function(
+                "Contact_force",
+                [ocp.symbolic_states, ocp.symbolic_controls],
+                [
+                    Dynamics.forces_from_forward_dynamics_torque_muscle_driven_with_contact(
+                        ocp.symbolic_states, ocp.symbolic_controls, nlp
+                    )
+                ],
+                ["x", "u"],
+                ["CS"],
+            ).expand()
+
+            n_contact = nlp["model"].nbContacts()
+            contacts_idx = PenaltyFunctionAbstract._check_and_fill_index(contacts_idx, n_contact, "contacts_idx")
+            data_to_track = PenaltyFunctionAbstract._check_and_fill_tracking_data_size(
+                data_to_track, [nlp["ns"], max(contacts_idx) + 1]
+            )
+
+            for i, v in enumerate(u):
+                force = CS_func(x[i], u[i])
+                val = force[contacts_idx] - data_to_track[t[i], contacts_idx]
+                penalty_type._add_to_penalty(ocp, nlp, val, **extra_param)
+
+            force = CS_func(x[-1], u[-1])
+            val = force[contacts_idx] - data_to_track[t[-1], contacts_idx]
+            penalty_type._add_to_penalty(ocp, nlp, val, **extra_param)
```

Comment:
This seems to assume that ALL is used. If it is the case, please make sure you raise an error if it is not the case. Or adjust this part when only START, MID, END of specific value is used